common/FormattingUtils.js

```diff
@@ -36,12 +36,19 @@ const formatWithMaxDecimals = (num, decimalPlaces, roundUp) => {
   }
 
   const fullPrecisionFloat = BigNumber(num);
-
-  // Convert back to BN to truncate any trailing 0s that the toFixed() output would print.
-  const fixedPrecisionFloat = BigNumber(fullPrecisionFloat)
-    .toFixed(decimalPlaces)
-    .toString();
-
+  let fixedPrecisionFloat;
+  // Convert back to BN to truncate any trailing 0s that the toFixed() output would print. If the number is larger than
+  // 10 then truncate to `decimalPlaces` number of decimal places. EG 999.999 -> 999.99. If the number is less than 10
+  // then truncate to 2 * decimalPlaces precision. EG: 0.0022183471 -> 0.002218.
+  if (fullPrecisionFloat.gt(BigNumber(10))) {
+    fixedPrecisionFloat = BigNumber(fullPrecisionFloat)
+      .toFixed(decimalPlaces)
+      .toString();
+  } else {
+    fixedPrecisionFloat = BigNumber(fullPrecisionFloat)
+      .toPrecision(decimalPlaces * 2)
```

Comment:
I also want to draw attention here: in the case that the number is below the threshold the formatted will introduce 2x the decimal points in precision. For example, a number as 0.0000198281 would become 0.00001982 with `decimalPlaces=2`. Does this seem reasonable?

Reply:
Hmmm, why are we multiplying the decimals for small values? Is this to try to give the user a specific number of significant figures? If so, why don't we just build a sig fig formatting function to do this?

The only reason I say this is that this function is supposed to be fairly generic, and this rule feels like it will work well for things where we want 4 significant figures, but probably won't really work for other use cases.

Reply:
So the idea with this is if the number is really small (less than 10) then we don't want to simply truncate the number. rather, we want to use the `. toPrecision` function to limit the number of precision within the output term.

I do agree with you that this looses some of the generality if the `decimalPlaces` value grows too large.

How would you suggest dealing with both big and small numbers in the same implementation?

Reply:
I've thought more about what you did here and I think there are two options that could make sense:
1. Change `decimalPlaces` to `minSigFigs`. All numbers must have at least `minSigFigs` significant figures. If there are more than `minSigFigs` digits before the decimal point, then truncate the number at the decimal point. For example, with `minSigFigs` at 2:
```
1234.123 -> 1234.
12.123 -> 12.
1.123 -> 1.1
0.1234 -> 0.12
0.00123 -> 0.0012
```

2. For numbers >= 1, interpret `numDecimals` as fixed and for numbers < 1, interpret `numDecimals` as sig figs. For example, with `numDecimals` at 2:
```
1234.123 -> 1234.12
12.123 -> 12.12
1.123 -> 1.12
0.1234 -> 0.12
0.00123 -> 0.0012
```

I think 2 would be easier to implement (pretty close to what you have now), so I would prefer that option, although it might be a little more confusing. I'd be curious to hear y'all's thoughts on these options @chrismaree @nicholaspai. Also, lmk if y'all have other strategies that y'all think generalize s.t. we could use them pretty much everywhere.

Reply:
Thanks for this feedback @mrice32 . I think your solution 2 is good but the problem here imo is that we lose the precision in small numbers which makes it not very useful. Representing `0.00123` as `0.0012` means losing half the significant figures. Take for example the current price of ETHBTC at `0.023456789012`. Under this this would be represented by `0.023`. In my opinion this is not enough precision. You would need to see at least 3 (or 4 ) significant figures to be able to see the true price. However, applying this 3 or 4 rule to a much larger number would look bad and results in too many trailing numbers after the 0.

After playing around with it a bunch and thinking more about it I think it best to have two configuration parameters: one defines how we deal with numbers `>1` (`decimalPlaces`) and the other defines how we deal with numbers `<1` (`minPrecision`). This way we can represent a whole range of numbers without loss of generality.

Let me know what you think :)

Reply:
SGTM!